Add I2C support for MCXN947

mmahadevan108 opened the pull request:
This was tested using `zephyr/tests/drivers/i2c/i2c_target_api`

mmahadevan108 committed:
drivers: i2c: MCUX SDK LPI2C driver sometimes uses instance number

Update the driver to account for variations in the SDK driver
when it uses the instance number instead of the base address.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
dts: nxp_lpi2c: Interrupts is no longer a required property

Some NXP SoC's have a FlexComm interface that manages the
interrupts.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
drivers: i2c: Use the NXP Flexcomm driver for interrupt handling

The Low Power Flexcomm driver manages the interrupt handling
and provides an API to register interrupt callbacks.
Register the NXP LPI2C interrupt handler.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
boards: frdm_mcxn947: Add support for I2C

Add support for I2C

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
tests: i2c: Add support for NXP MCXN947

Add support for NXP MCXN947

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
gwest.yml: Get the NXP SDK I2C support on MCX947

Get the NXP SDK driver for I2C support on MCXN947

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

danieldegrasse commented on boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0_defconfig:
```diff
@@ -7,6 +7,7 @@
 CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
 CONFIG_SERIAL=y
+CONFIG_MFD=y
```
Why is `CONFIG_MFD` being enabled by default here? Haven't we usually been selecting it from the LPI2C/LPUART/LPSPI drivers? Or is this required since LPUART and LPI2C can be enabled on the same LPFLEXCOMM?

mmahadevan108 replied:
I have addressed this by enabling it in the `soc` `Kconfig` whenever `LPFlexcomm` is enabled.

mmahadevan108 committed:
soc: nxp: Select the MFD Kconfig when LPFlexcomm is enabled

LPFlexcomm is a MFD device hence select this Kconfig whenever
it is enabled.
Remove the selection from the individual driver Kconfig files.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>
